pkgs/build-support/node/fetch-npm-deps/src/main.rs

```diff
@@ -255,13 +279,23 @@ fn main() -> anyhow::Result<()> {
         .unwrap()
         .into_par_iter()
         .try_for_each(|(dep, package)| {
-            if dep.is_empty() || package.resolved.is_none() {
+            if dep.is_empty()
+                || package.resolved.is_none()
+                || package
+                    .resolved
+                    .as_ref()
+                    .map(|r| matches!(r, UrlOrString::String(_)))
+                    .unwrap_or_default()
+            {
                 return Ok::<_, anyhow::Error>(());
             }
```

Comment:
Nit: I think this could read better if pulled out into two separate filter calls, one that filters out empty dependencies and one that filters out non-URL `resolved`. The latter one, in particular, I think is slicker reversing the conditional, something like:
```
.filter(|(_, package)| matches!(package.resolved, Some(UrlOrString::Url(_))))
```